src/SFA.DAS.ApprenticeCommitments.UnitTests/Infrastructure/SqlServerConnectionFactory/WhenConnectingToLocalDatabase.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Data.Common;
+using FluentAssertions;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.ApprenticeCommitments.Data.Models;
+
+namespace SFA.DAS.ApprenticeCommitments.UnitTests.Infrastructure.SqlServerConnectionFactory
+{
+    public class WhenConnectingToLocalDatabase
+    {
+        private ApprenticeCommitments.Infrastructure.SqlServerConnectionFactory _sut;
+        private Mock<IConfiguration> _configurationMock;
+        private DbContextOptionsBuilder<ApprenticeCommitmentsDbContext> _dbContextOptionsBuilder;
+        private string _connectionString;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _configurationMock = new Mock<IConfiguration>();
+            _dbContextOptionsBuilder = new DbContextOptionsBuilder<ApprenticeCommitmentsDbContext>();
+            _sut = new ApprenticeCommitments.Infrastructure.SqlServerConnectionFactory(_configurationMock.Object);
+            _connectionString = "Data Source=(localdb);Initial Catalog=DummyDatabase;Integrated Security=True";
+        }
+
+        [TestCase("LOCAL")]
+        [TestCase("ACCEPTANCE_TESTS")]
+        [TestCase("DEV")]
+        public void Then_CreateConnection_should_accept_a_local_connection_string(string environmentName)
+        {
+            _configurationMock.Setup(x => x[It.IsAny<string>()]).Returns(environmentName);
+
+            var dbConnection = _sut.CreateConnection(_connectionString);
+            dbConnection.Should().NotBeNull();
+        }
+
+        [Test]
+        public void Then_CreateConnection_should_throw_exception_because_connection_string_contains_integrated_security()
+        {
+            _configurationMock.Setup(x => x[It.IsAny<string>()]).Returns("TEST");
```

Comment:
The test name says "connection string contains integrated security", but the test itself simply says the configuration contains "TEST".  It's not clear to me what is being tested here.

Reply:
It's actually testing for an exception thrown when we try to add an AccessToken to a SQL connection string which say Integrated Security=True and this is automatically thrown by the AzureServiceTokenProvider

ie

Data Source=(localdb);Initial Catalog=DummyDatabase;Integrated Security=True"